metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1964,4 +1964,13 @@ public List<SQLPrimaryKey> getPrimaryKeys(String db_name, String tbl_name) throw
     }
     return ret;
   }
+
+  long updateTableParam(Table table, String key, String expectedValue, String newValue) {
+    String queryText = String.format("UPDATE \"TABLE_PARAMS\" SET \"PARAM_VALUE\" = ? " +
+            "WHERE \"PARAM_KEY\" = ? AND \"PARAM_VALUE\" = ? AND \"TBL_ID\" IN " +
```

Comment:
Just for reference, we hit performance issues in our production environment on executing this SQL. Changing the `IN` to `=` would solve this issue, at least for MySQL 5.7

<img width="1204" alt="image" src="https://github.com/user-attachments/assets/642c8be4-18ec-457f-82fa-7aeaf93f421d" />
<img width="1205" alt="image" src="https://github.com/user-attachments/assets/5c296295-1fdb-40aa-81d7-030ec2d148e6" />

Reply:
Thanks @pan3793 for reporting the issue. I agree it's better to use scalar subquery here. But since both 2.x and 3.x lines are EOL, and subquery is not used in 4.0, so I guess we won't have to do anything at the moment, right?

Reply:
Yes, the Hive 4.0 doesn't have such an issue. This is just a tip for users who want to port this patch to their internal legacy Hive version (2.3 or 3.x)